### PR TITLE
smacc2: 2.3.18-1 in 'humble/distribution.yaml' [bloom]

### PR DESCRIPTION
**Pull request by pabloinigoblasco**
Increasing version of package(s) in repository `smacc2` to `2.3.18-1`:

- upstream repository: https://github.com/robosoft-ai/SMACC2.git
- release repository: https://github.com/robosoft-ai/SMACC2-release.git
- distro file: `humble/distribution.yaml`
- bloom version: `0.11.2`
- previous version for package: `2.3.8-1`

## lifecyclenode_client

```
* minor
* Merge branch 'humble' of https://github.com/robosoft-ai/SMACC2 into humble
* Brettpac branch (#518 <https://github.com/robosoft-ai/SMACC2/issues/518>)
  * Attempt to fix weird issue with ros buildfarm
  * More on this buildfarm issue
  ---------
  Co-authored-by: brettpac <mailto:brettpac@pop-os.localdomain>
* Contributors: brettpac, pabloinigoblasco
```

## ros_timer_client

```
* Merge branch 'humble' of https://github.com/robosoft-ai/SMACC2 into humble
* Brettpac branch (#518 <https://github.com/robosoft-ai/SMACC2/issues/518>)
  * Attempt to fix weird issue with ros buildfarm
  * More on this buildfarm issue
  ---------
  Co-authored-by: brettpac <mailto:brettpac@pop-os.localdomain>
* Contributors: brettpac, pabloinigoblasco
```

## sm_atomic

```
* Merge branch 'humble' of https://github.com/robosoft-ai/SMACC2 into humble
* Brettpac branch (#518 <https://github.com/robosoft-ai/SMACC2/issues/518>)
  * Attempt to fix weird issue with ros buildfarm
  * More on this buildfarm issue
  ---------
  Co-authored-by: brettpac <mailto:brettpac@pop-os.localdomain>
* Contributors: brettpac, pabloinigoblasco
```

### CODE REVIEW

File: humble/distribution.yaml [hunk 1]
```diff
@@ -6885,12 +6885,15 @@ repositories:
       version: humble
     release:
       packages:
+      - lifecyclenode_client
```
Comment (emersonknapp) [inline humble/distribution.yaml:6888]: Are these package additions intentional? They do not adhere to REP-144 guidelines - but from what I can tell they are sample code within the SMACC repository, not intended for release.

**Review by emersonknapp — CHANGES_REQUESTED** [review]
Not sure about the added packages - waiting for response

File: humble/distribution.yaml (hunk 1 above)
Comment (brettpac) [reply humble/distribution.yaml:6888]: Yes, they are intentional. In what way do they not adhere to REP-144? These package additions are SMACC Client Behavior Libraries, and serve as modular adapters to different ROS systems. We have clients for Nav2, MoveIt, these two and a few more https://github.com/robosoft-ai/SMACC2/tree/humble/smacc2_client_library. These two offer functionality used by some sample state machines such as sm_atomic.

Comment (quarkytale) [reply humble/distribution.yaml:6888]: Looks good to me, not sure which naming rule was violated.

Comment (emersonknapp) [reply humble/distribution.yaml:6888]: `lifecyclenode_client` and `ros_timer_client` both sound really generic - but these subpackages are specific to `smacc2`, are they not? That's what I'm referring to about [REP-144](https://ros.org/reps/rep-0144.html)

> Package names should be specific enough to identify what the package does

Perhaps more appropriate names for these packages would use `smacc2` prefixing: `smacc2_lifecycle_client`, `smacc2_timer_client`

> A package name should not contain `ros` as it is redundant

This is my concern with `ros_timer_client`.

Comment (quarkytale) [reply humble/distribution.yaml:6888]: Makes sense, [nav2_lifecycle_manager ](https://index.ros.org/p/nav2_lifecycle_manager/github-ros-planning-navigation2/#humble) is probably a good example here.

Comment (pabloinigoblasco) [reply humble/distribution.yaml]: Hello. I introduced some changes: only smacc2 and smacc2_msgs are now distributed by the moment

**Review by emersonknapp — APPROVED** [review]

**Pull request merged**